aidatlu/constellation/aidatlu_satellite.py

```diff
@@ -0,0 +1,203 @@
+        logger._reset_all_loggers()
+        self.aidatlu.log = self.log
+        self.aidatlu.io_controller.log = self.log
+        self.aidatlu.dac_controller.log = self.log
+        self.aidatlu.clock_controller.log = self.log
+        self.aidatlu.trigger_logic.log = self.log
+        self.aidatlu.dut_logic.log = self.log
+        self.aidatlu.configure()
+        return "Do reconfigure complete"
+
+    def do_starting(self, run_identifier: str = None) -> str:
+        if self.use_mock:
+            start_time = time.time()
+
+            def _get_timestamp(self):
+                # Helper function returns timestamp
+                return (time.time() - start_time) / 25 * 1000000000
+
+            def _pull_fifo_event(self):
+                # Blank FIFO pull helper function
+                return 0
+
+            # Overwrite TLU methods needed for run loop
+            func_type = type(self.aidatlu.get_timestamp)
+            self.aidatlu.get_timestamp = func_type(_get_timestamp, self.aidatlu)
+            func_type = type(self.aidatlu.pull_fifo_event)
+            self.aidatlu.pull_fifo_event = func_type(_pull_fifo_event, self.aidatlu)
+
+        self.aidatlu.start_run_configuration()
+
+        return "Do starting complete"
+
+    def do_run(self, run_identifier: str) -> str:
+        t = threading.Thread(target=self.aidatlu.handle_status)
+        t.start()
+        while not self._state_thread_evt.is_set():
+            self.aidatlu.run_loop()
+        t.do_run = False
+        self.aidatlu.stop_run()
+        return "Do running complete"
+
+    def do_stopping(self) -> str:
+        return "Do stopping complete"
+
+    def do_landing(self) -> str:
+        return "Do Stop"
+
+    @schedule_metric("Hz", MetricsType.LAST_VALUE, 1)
+    def trigger_in_rate(self) -> Any:
+        if self.fsm.current_state_value == SatelliteState.RUN and hasattr(
+            self.aidatlu, "particle_rate"
+        ):
+            return self.aidatlu.particle_rate
+        else:
+            return None
+
+    @schedule_metric("Hz", MetricsType.LAST_VALUE, 1)
+    def trigger_out_rate(self) -> Any:
+        if self.fsm.current_state_value == SatelliteState.RUN and hasattr(
+            self.aidatlu, "hit_rate"
+        ):
+            return self.aidatlu.hit_rate
+        else:
+            return None
+
+    @schedule_metric("", MetricsType.LAST_VALUE, 1)
+    def trigger_total_trigger_nr(self) -> Any:
+        if self.fsm.current_state_value == SatelliteState.RUN and hasattr(
+            self.aidatlu, "total_trigger_number"
+        ):
+            return self.aidatlu.total_trigger_number
+        else:
+            return None
+
+    @schedule_metric("", MetricsType.LAST_VALUE, 1)
+    def sc0(self) -> Any:
+        if self.fsm.current_state_value == SatelliteState.ORBIT:
+            self.log.debug("sc0: %s" % self.aidatlu.get_scalar(0))
+            return self.aidatlu.get_scalar(0)
```

Comment:
I think I missed that in another PR...it should be `scaler`, not `scalar`. Maybe we need to make a fix and rebase.